feature_extraction.py

```diff
@@ -140,3 +140,206 @@ def transform(self, X):
                         how='left')
 
         return {'numerical_features': X[self.groupby_aggregations_names].astype(np.float32)}
+
+
+class Application(BaseTransformer):
+    def __init__(self):
+        super().__init__()
+        self.application_names = ['ANNUITY_INCOME_PERCENTAGE',
+                                  'CREDIT_TO_GOODS_RATIO',
+                                  'DAYS_EMPLOYED_PERCENTAGE',
+                                  'EXT_SOURCES_MEAN',
+                                  'INCOME_CREDIT_PERCENTAGE',
+                                  'INCOME_PER_PERSON',
+                                  'PAYMENT_RATE']
+
+    def transform(self, X, y=None):
+        X['DAYS_EMPLOYED'].replace(365243, np.nan, inplace=True)
+        # SIMPLE
+        X['ANNUITY_INCOME_PERCENTAGE'] = X['AMT_ANNUITY'] / X['AMT_INCOME_TOTAL']
+        X['CREDIT_TO_GOODS_RATIO'] = X['AMT_CREDIT'] / X['AMT_GOODS_PRICE']
+        X['DAYS_EMPLOYED_PERCENTAGE'] = X['DAYS_EMPLOYED'] / X['DAYS_BIRTH']
+        X['EXT_SOURCES_MEAN'] = X[['EXT_SOURCE_1', 'EXT_SOURCE_2', 'EXT_SOURCE_3']].mean(axis=1)
+        X['INCOME_CREDIT_PERCENTAGE'] = X['AMT_INCOME_TOTAL'] / X['AMT_CREDIT']
+        X['INCOME_PER_PERSON'] = X['AMT_INCOME_TOTAL'] / X['CNT_FAM_MEMBERS']
+        X['PAYMENT_RATE'] = X['AMT_ANNUITY'] / X['AMT_CREDIT']
+
+        return {'numerical_features': X[self.application_names]}
+
+
+class Bureau(BaseTransformer):
+    def __init__(self, filepath, id_columns, **kwargs):
+        self.filepath = filepath
+        self.id_columns = id_columns
+        self.bureau_names = ['bureau_active_loans_percentage',
+                             'bureau_average_creditdays_prolonged',
+                             'bureau_average_enddate_future',
+                             'bureau_average_loan_type',
+                             'bureau_credit_enddate_percentage',
+                             'bureau_days_credit_diff',
+                             'bureau_debt_credit_ratio',
+                             'bureau_loan_count',
+                             'bureau_loan_types',
+                             'bureau_overdue_debt_ratio'
+                             ]
+
+    def fit(self, X):
+        bureau = pd.read_csv(self.filepath)
```

Comment:
@pknut I don't like that we read data in the fit method. It is very NOT single responsibility principle. I don't mind reading data in a separate transformer but I would rather read them in the `main.py` and pass objects to pipeline